packages/core/src/components/Picker/Picker.tsx

```diff
@@ -60,39 +123,305 @@ const Picker: React.FC<Props> = ({
+    ...{
+      height: 60,
+      width: "100%",
+    },
+    ...omit(viewStyles, [...borders, ...margins]),
+    ...{ maxHeight },
+  };
+
+  const selectedLabel =
+    internalValue &&
+    ((pickerOptions as unknown as PickerOption[]).find(
+      (option) => option.value === internalValue
+    )?.label ??
+      internalValue);
+
+  const labelText = label ? (
+    <Text
+      style={{
+        textAlign: textStyles.textAlign,
+        color: unstyledColor,
+        fontSize: 12,
+        paddingBottom: 4,
+      }}
+    >
+      {label}
+    </Text>
+  ) : null;
+
+  const leftIconOutset = leftIconMode === "outset";
+
+  const leftIcon = leftIconName ? (
+    <Icon
+      name={leftIconName}
+      color={disabled ? unstyledColor : iconColor}
+      size={iconSize}
+      style={{
+        marginRight: 4,
+        marginLeft: 4,
+      }}
+    />
+  ) : null;
+
+  const rightIcon = rightIconName ? (
+    <Icon
+      name={rightIconName}
+      color={disabled ? unstyledColor : iconColor}
+      size={iconSize}
+      style={{
+        marginRight: -10,
+        marginLeft: 8,
+      }}
+    />
+  ) : null;
+
+  const width = stylesWithoutBordersAndMargins?.width ?? undefined;
+
+  const textAlign = textStyles?.textAlign;
+
+  const paddingLeft =
+    leftIconOutset &&
+    (!textAlign || textAlign === "left" || textAlign === "justify")
+      ? iconSize + 4
+      : 0;
+
+  const assistiveTextLabel = assistiveText ? (
+    <Text
+      style={{
+        textAlign,
+        width,
+        paddingLeft,
+        color: unstyledColor,
+        fontSize: 12,
+        paddingTop: 4,
+      }}
+    >
+      {assistiveText}
+    </Text>
+  ) : null;
+
+  const primaryTextStyle = {
+    color: unstyledColor,
+    fontSize: 14,
+    ...pickBy(textStyles, identity),
+    ...(placeholder === internalValue ? { color: placeholderTextColor } : {}),
+    ...(disabled ? { color: unstyledColor } : {}),
+  };
+
   const handleValueChange = (newValue: string, itemIndex: number) => {
-    setInternalValue(newValue);
-    if (onValueChange) {
-      onValueChange(newValue, itemIndex);
+    if (!placeholder || itemIndex > 0) {
+      onValueChange?.(newValue, itemIndex);
     }
+    setInternalValue(newValue);
   };
 
   return (
-    <PickerComponent
-      {...props}
-      selectedValue={String(internalValue)}
-      placeholder={placeholder}
-      options={pickerOptions}
-      onValueChange={handleValueChange}
-    />
+    /* marginsContainer */
+    <View style={[styles.marginsContainer, marginStyles]}>
+      {/* touchableContainer */}
+      <Touchable
+        disabled={disabled}
+        onPress={togglePickerVisible}
+        style={styles.touchableContainer}
+      >
+        {/* outsetContainer */}
+        <View
+          pointerEvents="none"
+          style={[
+            styles.outsetContainer,
+            stylesWithoutBordersAndMargins,
+            // @ts-expect-error ... rejects border-related { key: undefined }
```

Comment:
what's `{key: undefined}`? am i misunderstanding that?

Reply:
@peterpme Removed comment / Added type / N/A issue.